lib/core/src/animation/easing.rs

```diff
@@ -0,0 +1,122 @@
+//! This module implements easing formulas and related utilities. To learn more about easing
+//! functions, please refer to the following link: https://easings.net/en
+
+// TODO: It would be useful to make the easing functions polymorphic. Maybe we can make use of
+// nalgebra::RealField?
+
+use core::f32::consts::PI;
+
+/// Easing function signature.
+pub trait FnEasing = Fn(f32) -> f32 + 'static;
+
+
+
+// ========================
+// === Easing functions ===
+// ========================
+// Reference: http://easings.net/en
+
+macro_rules! easing_fn {
+    (pub fn $name:ident(t:f32) -> f32 $block:block) => { paste::item! {
```

Comment:
Why don't you write a macro `easing_fns` and just put all functions in it instead of using this macro so often below? :)